src/client/datascience/notebook/helpers/helpers.ts

```diff
@@ -814,3 +814,9 @@ export async function updateVSCNotebookAfterTrustingNotebook(
         });
     });
 }
+
+export function findAssociatedNotebookDocument(cellUri: Uri, vscodeNotebook: IVSCodeNotebook) {
+    return vscodeNotebook.notebookDocuments.find((item) =>
+        item.cells.some((cell) => cell.uri.toString() === cellUri.toString())
```

Comment:
Should be the file path comparison?

Reply:
Doesn't matter here as these are VSCode generated Uris (not from us). hence didn't bother with that.
But will add so that we're consistent.